tests/pkg_qr-code-generator/Makefile.board.dep

```diff
@@ -0,0 +1,5 @@
+# Boards with a screen can use disp_dev
```

Comment:
These extra dependencies could be expressed in a `Kconfig` file in the application, by something like:
```
config APPLICATION
    bool
    default y
    imply MODULE_AUTO_INIT_SCREEN if BOARD_HAS_DISPLAY
    imply MODULE_DISP_DEV if BOARD_HAS_DISPLAY
    depends on TEST_KCONFIG

config BOARD_HAS_DISPLAY
    bool
    default y
    depends on BOARD_PINETIME || BOARD_ADAFRUIT_CLUE || BOARD_STM32F429I_DISC1 || BOARD_STM32F429I_DISCO
```
To avoid running Kconfig by default this can be added in the Makefile
```make
# As there is an 'Kconfig' we want to explicitly disable Kconfig by setting
# the variable to empty
SHOULD_RUN_KCONFIG ?=
```